modules/virtual-cluster/main.tf

```
@@ -269,9 +269,11 @@ resource "aws_iam_role_policy_attachment" "additional" {
 resource "aws_cloudwatch_log_group" "this" {
   count = var.create && var.create_cloudwatch_log_group ? 1 : 0
 
-  name              = "/emr-on-eks-logs/emr-workload/${local.namespace}"
+  name              = var.cloudwatch_log_group_use_name_prefix ? null : coalesce(var.cloudwatch_log_group_name, "/emr-on-eks-logs/emr-workload/${local.namespace}")
+  name_prefix       = var.cloudwatch_log_group_use_name_prefix ? coalesce(var.cloudwatch_log_group_name_prefix, "/emr-on-eks-logs/emr-workload/${local.namespace}/") : null
```

and this becomes:

```suggestion
  name_prefix       = var.cloudwatch_log_group_use_name_prefix ? "${local.cloudwatch_log_group_name}-" : null
```

Similar to [this](https://github.com/terraform-aws-modules/terraform-aws-ecs/blob/d764aae10c537a381f6098587e877edc0c26798b/modules/service/main.tf#L430-L431)